src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -232,14 +233,34 @@ public void Dispose()
                     if (_identities.Count == 0)
                         return;
 
-                    _database.TxMerger.Enqueue(new UpdateIdentitiesCommand(_identities, _database)).Wait();
+                    _database.TxMerger.Enqueue(new UpdateClusterIdentitiesCommand(_identities, _database)).Wait();
                 }
                 finally
                 {
                     _returnContext?.Dispose();
                 }
             }
 
+            private class UpdateClusterIdentitiesCommand : TransactionOperationsMerger.MergedTransactionCommand
+            {
+                private readonly Dictionary<string, long> _identities;
+                private readonly DocumentDatabase _database;
+
+                public UpdateClusterIdentitiesCommand(Dictionary<string, long> identities, DocumentDatabase database)
+                {
+                    _identities = identities;
+                    _database = database;
+                }
+
+                public override int Execute(DocumentsOperationContext context)
+                {
+                    var (etag,_) = _database.ServerStore.SendToLeaderAsync(new UpdateClusterIdentityCommand(_database.Name,_identities)).Result;
```

Comment:
You are _inside the transaction merger_, making a _raft consensus call_. That is not a good idea!